src/components/CustomInput/index.tsx

```diff
@@ -1,53 +1,77 @@
-import { Input } from '@openware/components';
+import { InputGroup, FormControl } from 'react-bootstrap';
 import * as React from 'react';
 
 export interface CustomInputProps {
     type: string;
     label: string;
     defaultLabel: string;
-    handleChangeInput: (value: string) => void;
+    handleChangeInput?: (value: string) => void;
     inputValue: string;
-    handleFocusInput: () => void;
-    classNameLabel: string;
-    classNameInput: string;
+    handleFocusInput?: () => void;
+    classNameLabel?: string;
+    classNameInput?: string;
     placeholder: string;
     autoFocus?: boolean;
     onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+    readOnly?: boolean;
+    id?: string;
+    handleClick?: ((event: React.MouseEvent<HTMLInputElement, MouseEvent>) => void) | undefined;
+    isDisabled?: boolean | undefined;
 }
 
-class CustomInput extends React.Component<CustomInputProps> {
+interface OnChangeEvent {
+    target: {
+        value: string;
+    };
+}
+type Props = CustomInputProps;
+
+class CustomInput extends React.Component<Props> {
     public render() {
         const {
             label,
             placeholder,
             defaultLabel,
             inputValue,
             classNameLabel,
-            classNameInput,
             type,
             autoFocus,
-            onKeyPress,
+            readOnly,
+            id,
+            handleClick,
+            isDisabled,
         } = this.props;
 
         return (
             <React.Fragment>
-                <label className={classNameLabel}>
-                    {inputValue && (label || defaultLabel)}
-                </label>
-                <Input
-                    type={type}
-                    value={inputValue}
-                    placeholder={placeholder}
-                    className={classNameInput}
-                    onFocus={this.props.handleFocusInput}
-                    onBlur={this.props.handleFocusInput}
-                    onChangeValue={this.props.handleChangeInput}
-                    autoFocus={autoFocus}
-                    onKeyPress={onKeyPress}
-                />
+                <div className="custom-input">
+                    <label className={classNameLabel}>
+                        {inputValue && (label || defaultLabel)}
+                    </label>
+                    <InputGroup size="lg">
+                        <FormControl
+                            size="lg"
+                            type={type}
+                            value={inputValue}
+                            placeholder={placeholder}
+                            autoFocus={autoFocus}
+                            onFocus={this.props.handleFocusInput}
+                            onBlur={this.props.handleFocusInput}
+                            onChange={e => this.handleChangeValue(e)}
```

Comment:
please remove arrow function here, put props onChange={this.handleChangeValue}

Reply:
this arrow function resolve type conflict due to react-bootstrap source code